### ISSUE TITLE
5.0.1 is not compatible with other libp2ps

### ISSUE THREAD

**Issue by Menduist** [ChainSafe/js-libp2p-noise#127]
Trying to connect to nim-libp2p or go-libp2p using js-libp2p-noise v5.0.1:
```
AggregateError:
    Error: Error occurred during XX handshake: xx handshake stage 1 validation fail
        at Noise.performXXHandshake (./js-libp2p/node_modules/@chainsafe/libp2p-noise/dist/src/noise.js:185:19)
        at processTicksAndRejections (node:internal/process/task_queues:96:5)
        at async Noise.performHandshake (./js-libp2p/node_modules/@chainsafe/libp2p-noise/dist/src/noise.js:154:20)
        at async Noise.secureOutbound (./js-libp2p/node_modules/@chainsafe/libp2p-noise/dist/src/noise.js:79:27)
        at async Upgrader._encryptOutbound (./js-libp2p/src/upgrader.js:406:12)
        at async Upgrader.upgradeOutbound (./js-libp2p/src/upgrader.js:176:11)
        at async ClassIsWrapper.dial (./js-libp2p/node_modules/libp2p-websockets/src/index.js:54:18)
        at async TransportManager.dial (./js-libp2p/src/transport-manager.js:111:14)
        at async ./js-libp2p/src/dialer/dial-request.js:78:18
        at async ./js-libp2p/node_modules/p-some/index.js:53:19
    at maybeSettle (./js-libp2p/node_modules/p-some/index.js:31:11)
    at ./js-libp2p/node_modules/p-some/index.js:69:23
```

Works fine with 5.0.0

**Comment by dapplion** [comment]
CC @mpetrunic @wemeetagain

**Comment by wemeetagain** [comment]
It looks like this problem was introduced in #125 

I've got a branch that tests against go-libp2p showing the error: https://github.com/ChainSafe/js-libp2p-noise/tree/reproduce-127
The test needs polish, it not reliable and it doesn't properly shut down the go-libp2p daemon, but it proves the point. Testing against v5.0.0 shows success whereas testing against #125 shows the error.

@achingbrain any thoughts on what might have gone wrong?

**Comment by mpetrunic** [comment]
Fixed in [5.0.2](https://github.com/ChainSafe/js-libp2p-noise/releases/tag/v5.0.2)

**Comment by wemeetagain** [comment]
resolved with #129 and v5.0.2
